riscv/processor.h

```diff
@@ -265,8 +274,10 @@ class processor_t : public abstract_device_t
   }
   extension_t* get_extension() { return ext; }
   bool supports_extension(unsigned char ext) {
-    if (ext >= 'a' && ext <= 'z') ext += 'A' - 'a';
-    return ext >= 'A' && ext <= 'Z' && ((state.misa >> (ext - 'A')) & 1);
+    if (isupper(ext))
+      return ((state.misa >> (ext - 'A')) & 1);
+    else
+      return extension_table[ext];
```

Comment:
It would be more robust to do a bounds check here.  You can add an exta entry in the enum, `EXT_MAX`, and check if `ext < EXT_MAX` before indexing the table, and otherwise `abort()`.  The compiler will optimize this, since `supports_extension` is usually called with a constant argument.

Reply:
The table size is 256 equaling to the size of argument type "unsigned char"

Reply:
Oh, I missed that detail. Sounds good.